autogluon/task/text_classification/dataset.py

```diff
@@ -119,28 +122,167 @@ def get_dataset(self, segment='train'):
+    comment : str, optional
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    encoding : str, optional
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    dialect : str or csv.Dialect, optional
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    error_bad_lines : bool, default True
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    warn_bad_lines : bool, default True
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    delim_whitespace : bool, default False
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    low_memory : bool, default True
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    memory_map : bool, default False
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    float_precision : str, optional
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    """
+    def __init__(self, *args, **kwargs):
+        self._read(**kwargs)
+        self.is_pair = False
+        self.class_labels = list(set([sample[1] for sample in self.dataset]))
+        self.metric = Accuracy()
+        super(CustomTSVClassificationTask, self).__init__(self.class_labels, self.metric, self.is_pair)
+
+    def get_dataset(self, segment='train'):
+        """Get the corresponding dataset for the task.
+
+        Parameters
+        ----------
+        segment : str, default 'train'
+            Dataset segments.
+
+        Returns
+        -------
+        The dataset of target segment.
+        """
+        if segment == 'train':
+            return SampledDataset(self.dataset, self.train_sampler)
+        elif segment == 'dev':
+            return SampledDataset(self.dataset, self.dev_sampler)
+        else:
+            raise NotImplementedError
+
+    def _read(self, **kwargs):
+        path = kwargs.get('filepath', None)
+        kwargs['filepath_or_buffer'] = path
+        kwargs.pop("filepath")
+        dataset_df = pd.read_csv(**kwargs)
+        dataset_df_lst = dataset_df.values.tolist()
+        self.dataset = gluon.data.SimpleDataset(dataset_df_lst)
+        split_ratio = kwargs.get('split_ratio', None) if 'split_ratio' in kwargs else 0.8
+        self.train_sampler, self.dev_sampler = get_split_samplers(self.dataset, split_ratio=split_ratio)
 
     def dataset_train(self):
-        return 'train', self.trainset
+        """Get the training segment of the dataset for the task.
```

Comment:
why we need a function to do it?

Reply:
to make it align with the bert family training pipeline, see https://github.com/awslabs/autogluon/pull/363#discussion_r394675221

Reply:
Why does the bert family training pipeline need a function for this?

Reply:
For the data transformation in the pipeline, to align with the processing data functionality we have for the benchmarking bert family model as in the implementation.